crates/cubecl-linalg/src/convolution/algorithm/simple_tma.rs

```diff
@@ -152,28 +168,48 @@ pub(crate) fn check_problem_tma(problem: &ConvolutionProblem) -> Result<(), Inva
         }
     }
 
-    let corner = calculate_upper_corner(problem.padding, problem.kernel_size, problem.dilation);
-    check_range(corner[0] as isize, "corner_h", -128, 127)?;
-    check_range(corner[1] as isize, "corner_w", -128, 127)?;
+    let (corner_min, corner_max) = match problem.dimensionality {
+        Dimensionality::Dim1 => (-(2isize.pow(15)), 2isize.pow(15) - 1),
+        Dimensionality::Dim2 => (-(2isize.pow(7)), 2isize.pow(7) - 1),
+        Dimensionality::Dim3 => (-(2isize.pow(4)), 2isize.pow(4) - 1),
+    };
+
+    let corner = calculate_upper_corner(&problem.padding, &problem.kernel_size, &problem.dilation);
+    for (i, offs) in corner.iter().enumerate() {
+        check_range(
+            *offs as isize,
+            &format!("corner[{i}]"),
```

Comment:
You could avoid allocating a string just for the checks I guess. Using a function `|| format!("corner[{i}]")`